platform/files.go

```diff
@@ -85,17 +86,88 @@ type BuildMetadata struct {
 	PlatformAPI                 *api.Version             `toml:"-" json:"-"`
 }
 
+// DecodeBuildMetadataTOML reads a metadata.toml file
+func DecodeBuildMetadataTOML(path string, platformAPI *api.Version, buildmd *BuildMetadata) error {
```

Comment:
Could we please add a test for this function? Also this logic seems duplicated across `launch.DecodeLaunchMetadataTOML` and `buildpack.DecodeLaunchTOML` - maybe we could move the common logic to an internal package? 

The `launch` and `buildpack` functions seem not used outside the package, maybe they could be made private (this would avoid the need to test them too as they are already tested implicitly in the other units).

Reply:
[01eddfa](https://github.com/buildpacks/lifecycle/pull/900/commits/01eddfa4d00876c9d23fc61217601446c1596a9e)